srml/contract/src/lib.rs

```diff
@@ -14,41 +14,82 @@
 // You should have received a copy of the GNU General Public License
 // along with Substrate. If not, see <http://www.gnu.org/licenses/>.
 
-//! Smart-contract module for runtime; Allows deployment and execution of smart-contracts
-//! expressed in WebAssembly.
+//! # Contract Module
+//! 
+//! The contract module provides functionality for the runtime to deploy and execute WebAssembly smart-contracts.
+//! The supported dispatchable functions are documented as part of the [`Call`](./enum.Call.html) enum.
+//! 
+//! ## Overview
+//! 
+//! This module extends accounts (see `Balances` module) to have smart-contract functionality.
+//! These "smart-contract accounts" have the ability to create smart-contracts and make calls to other contract 
+//! and non-contract accounts.
+//! 
+//! The smart-contract code is stored once in a code cache, and later retrievable via a code hash.
```

Comment:
```suggestion
//! The smart-contract code is stored once in a code cache, and later retrievable via that code hash.
```
This comes across more clear?

Reply:
ah ill change to 'via its `codehash`'. every code cache has just one code hash